### ISSUE TITLE
Cannot compile pcl 1.8 on Ubuntu 14.04

### ISSUE THREAD

**Issue by mgarbade** [PointCloudLibrary/pcl#1881]
:warning: This is a issue tracker, please use our mailing list for questions: www.pcl-users.org. :warning: 

upon doing cmake I get a warning:
 ```
CMake Warning at cmake/pcl_targets.cmake:864 (message):
   OpenNI grabber support: not building because OpenNI not found
 Call Stack (most recent call first):
   CMakeLists.txt:297 (PCL_ADD_GRABBER_DEPENDENCY)



 CMake Warning at cmake/pcl_targets.cmake:864 (message):
   OpenNI2 grabber support: not building because OpenNI2 not found
 Call Stack (most recent call first):
   CMakeLists.txt:298 (PCL_ADD_GRABBER_DEPENDENCY)



 CMake Warning at cmake/pcl_targets.cmake:864 (message):
   Fotonic camera support: not building because FZAPI not found
 Call Stack (most recent call first):
   CMakeLists.txt:299 (PCL_ADD_GRABBER_DEPENDENCY)



 CMake Warning at cmake/pcl_targets.cmake:864 (message):
   IDS-Imaging Ensenso camera support: not building because Ensenso not found
 Call Stack (most recent call first):
   CMakeLists.txt:300 (PCL_ADD_GRABBER_DEPENDENCY)



 CMake Warning at cmake/pcl_targets.cmake:864 (message):
   David Vision Systems SDK support: not building because davidSDK not found
 Call Stack (most recent call first):
   CMakeLists.txt:301 (PCL_ADD_GRABBER_DEPENDENCY)



 CMake Warning at cmake/pcl_targets.cmake:864 (message):
   RealSense SDK support: not building because RSSDK not found
 Call Stack (most recent call first):
   CMakeLists.txt:303 (PCL_ADD_GRABBER_DEPENDENCY)
```
and if I ignore them even more warnings (still in cmake):

 ```
CMake Warning at cmake/pcl_targets.cmake:194 (add_library):
   Cannot generate a safe runtime search path for target pcl_io because files
   in some directories may conflict with libraries in implicit directories:

     runtime library [libusb-1.0.so.0] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
       /opt/softkinetic/DepthSenseSDK/lib

   Some of these libraries may not be found correctly.
 Call Stack (most recent call first):
   io/CMakeLists.txt:322 (PCL_ADD_LIBRARY)



 CMake Warning at cmake/pcl_targets.cmake:263 (add_executable):
   Cannot generate a safe runtime search path for target
   pcl_convert_pcd_ascii_binary because files in some directories may conflict
   with libraries in implicit directories:

     runtime library [libusb-1.0.so.0] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
       /opt/softkinetic/DepthSenseSDK/lib

   Some of these libraries may not be found correctly.
 Call Stack (most recent call first):
   io/tools/CMakeLists.txt:15 (PCL_ADD_EXECUTABLE)



 CMake Warning at cmake/pcl_targets.cmake:263 (add_executable):
   Cannot generate a safe runtime search path for target pcl_converter because
   files in some directories may conflict with libraries in implicit
   directories:

     runtime library [libusb-1.0.so.0] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
       /opt/softkinetic/DepthSenseSDK/lib

   Some of these libraries may not be found correctly.
 Call Stack (most recent call first):
   io/tools/CMakeLists.txt:17 (PCL_ADD_EXECUTABLE)



 CMake Warning at cmake/pcl_targets.cmake:263 (add_executable):
   Cannot generate a safe runtime search path for target pcl_hdl_grabber
   because files in some directories may conflict with libraries in implicit
   directories:

     runtime library [libusb-1.0.so.0] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
       /opt/softkinetic/DepthSenseSDK/lib

   Some of these libraries may not be found correctly.
 Call Stack (most recent call first):

```
Then after calling make, it fails with:

```
[ 28%] Linking CXX executable ../bin/pcl_compute_hausdorff
[ 28%] Linking CXX executable ../bin/pcl_vtk2ply
[ 28%] Linking CXX executable ../bin/pcl_pcd2vtk
../lib/libpcl_kdtree.so.1.8.0: undefined reference to `LZ4_resetStreamHC'
../lib/libpcl_kdtree.so.1.8.0: undefined reference to `LZ4_setStreamDecode'
../lib/libpcl_kdtree.so.1.8.0: undefined reference to `LZ4_decompress_safe'
../lib/libpcl_kdtree.so.1.8.0: undefined reference to `LZ4_decompress_safe_continue'
../lib/libpcl_kdtree.so.1.8.0: undefined reference to `LZ4_compress_HC_continue'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/pcl_compute_hausdorff] Error 1
make[1]: *** [tools/CMakeFiles/pcl_compute_hausdorff.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 28%] Building CXX object surface/CMakeFiles/pcl_surface.dir/src/convex_hull.cpp.o
[ 28%] Built target pcl_vtk2ply
[ 28%] Linking CXX executable ../bin/pcl_tiff2pcd
[ 29%] Building CXX object surface/CMakeFiles/pcl_surface.dir/src/3rdparty/poisson4/factor.cpp.o
[ 29%] Building CXX object surface/CMakeFiles/pcl_surface.dir/src/3rdparty/poisson4/geometry.cpp.o
[ 29%] Building CXX object surface/CMakeFiles/pcl_surface.dir/src/3rdparty/poisson4/marching_cubes_poisson.cpp.o
[ 29%] Built target pcl_pcd2vtk
[ 29%] Built target pcl_tiff2pcd
[ 29%] Linking CXX executable ../bin/pcl_pclzf2pcd
[ 29%] Linking CXX executable ../bin/pcl_concatenate_points_pcd
[ 29%] Built target pcl_pclzf2pcd
[ 29%] Built target pcl_concatenate_points_pcd
[ 29%] Linking CXX shared library ../lib/libpcl_surface.so
[ 29%] Built target pcl_surface
Scanning dependencies of target pcl_gpu_features
[ 30%] Building CXX object gpu/features/CMakeFiles/pcl_gpu_features.dir/src/features.cpp.o
[ 30%] Linking CXX shared library ../../lib/libpcl_gpu_features.so
[ 30%] Built target pcl_gpu_features
make: *** [all] Error 2

```

## My Environment
* Operating System and version: Ubuntu 14.04
* Titan X Pascal
* Cuda_8.0.61
* Compiler: gcc 4.8.4
* PCL Version: 1.8

## Other remarks
I already tried installing via ppa  
`    sudo add-apt-repository ppa:v-launchpad-jochen-sprickerhof-de/pcl`
but also apt-get could not install pcl:
```
The following packages have unmet dependencies:
 libpcl-all : Depends: libpcl-1.7-all but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```
I already installed tons of requirements but the problem persists
- VTK (fresh from github --> is there any special version I need, maybe the newest is not compatible?)
- FLANN (both from github and via apt-get)

Any idea what goes wrong? (I've already wasted half a day)

**Comment by taketwo** [comment]
This is a issue tracker, please use our mailing list for questions: www.pcl-users.org.
The warnings you get are as expected. From the build output I'd say you have problems with your flann installation. Possibly related #804.

**Comment by mgarbade** [comment]
That was the trick! Thanks a lot! (I wasted hours on that sh**)
I kicked out the manually compiled flann ( from git) and instead installed the system version
    `sudo apt-get install libflann-dev`
Then the compilation worked fine.

Other packages I installed (but were unrelated to my bug / probably unnecessary) - just for completeness:

`sudo apt-get install g++ cmake cmake-gui doxygen   mpi-default-dev openmpi-bin openmpi-common libeigen3-dev libboost-all-dev libvtk5.8-qt4 libvtk5.8 libqhull* libusb-dev libgtest-dev git-core freeglut3-dev pkg-config build-essential libxmu-dev libxi-dev  libusb-1.0-0-dev graphviz mono-complete qt-sdk libeigen3-dev`
